src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3537,9 +3537,11 @@ static TypeWithState applyPostConditionsWhenTrue(TypeWithState typeWithState, Fl
             {
                 bool notNullWhenTrue = (annotations & FlowAnalysisAnnotations.NotNullWhenTrue) != 0;
                 bool maybeNullWhenTrue = (annotations & FlowAnalysisAnnotations.MaybeNullWhenTrue) != 0;
+                bool maybeNullWhenFalse = (annotations & FlowAnalysisAnnotations.MaybeNullWhenFalse) != 0;
 
-                if (maybeNullWhenTrue)
+                if (maybeNullWhenTrue && !(maybeNullWhenFalse && notNullWhenTrue))
                 {
+                    // [MaybeNull, NotNullwhen(true)] means [MaybeNullWhen(false)]
```

Comment:
>NotNullwhen [](start = 35, length = 11)

Nit: NotNullWhen

Reply:
And below.

---
In reply to: [302791634](https://github.com/dotnet/roslyn/pull/37159#discussion_r302791634) [](ancestors = 302791634)